rfcs/0079-include-key-as.md

```diff
@@ -0,0 +1,94 @@
+  `key encode` > infer from `format ... encode ...` > default type  
+* **Important**: `include key` is required for `format upsert` and RisingWave will use the key column as one and 
+  only primary key to perform upsert semantic. It does not allow to specify multiple columns as primary key
+  even if they are part of the key.
+
+## Specifications
+
+### Kafka
+
+| Allowed Components | Default Type                               | Note                                              |
+|--------------------|--------------------------------------------|---------------------------------------------------|
+| key                | `bytea`                                    | Allow overwritten by `encode` and `key encode`    |
+| timestamp          | `timestamp with time zone` (i64 in millis) | Refer to `CreateTime` rather than `LogAppendTime` |
+| partition          | `varchar`                                      | The message is from which partition               |
+| offset             | `varchar`                                      | The offset in the partition                       |
+| header             | `struct<varchar, bytea>[]`                 | KV pairs along with message                       |
+
+### Pulsar
+
+| Allowed Components | Default Type | Note                                                                                      |
+|--------------------|--------------|-------------------------------------------------------------------------------------------|
+| key                | `bytea`      | Allow overwritten by `encode` and `key encode`. Refer to `MessageMetadata::partition_key` |
+| partition          | `varchar`        | The message is from which partition                                                       |
+| offset             | `varchar`        | The offset in the partition                                                               |
+
+More components are available at [here](https://docs.rs/pulsar/latest/pulsar/message/proto/struct.MessageMetadata.html).
+
+### Kinesis
+
+| Allowed Components | Default Type                             | Note                                                                             |
+|--------------------|------------------------------------------|----------------------------------------------------------------------------------|
+| key                | `bytea`                                  | Allow overwritten by `encode` and `key encode`. Refer to `Record::partition_key` |
+| timestamp          | `timestamp with time zone` (from chrono) | refer to `Record::approximate_arrival_timestamp`                                 |
+| partition          | `varchar`                                    | The message is from which partition                                              |
+| offset             | `varchar`                                    | The offset in the partition                                                      |
```

Comment:
```suggestion
| offset             | `varchar`                                    | The offset in the partition. Corresponds to Kinesis sequence numbers.                                                     |
```

Users might feel confused about the concept's name. At least we might need to document it clearly.

Reply:
BTW, I'm thinking that whether the default name can be different, even if we want the unified data type.

I think we will rely on `AdditionalColumnType` when the column is needed, instead of the col name.

Reply:
> BTW, I'm thinking that whether the default name can be different, even if we want the unified data type.

of course, the `offset`/`partition` here serves as an identifier where Risingwave loads the field. The default name should be `_rw_{connector name}_offset`.
And there is an `as` clause following the identifier, allowing renaming the column. 

> I think we will rely on AdditionalColumnType when the column is needed, instead of the col name.

Yes, it exactly what I design